evenDistributionScore.py

```diff
@@ -64,12 +63,131 @@ def breakPattern(length):
         section = np.linspace(int(score[-1]), endpoint, num)
         score = np.concatenate((score, section))
 
-    score = np.concatenate((score, np.zeros(fiveMins))) # add final 5 minutes of zero score
+    score = np.concatenate((score, np.zeros(fiveMins)))  # add final 5 minutes of zero score
 
     return score
 
-        
-# Plot distibution to visualise
-x=np.linspace(0, getLength(), getLength())
+#Plot distibution to visualise
+x = np.linspace(0, getLength(), getLength())
 plt.plot(x, breakPattern(getLength()))
 plt.show()
+
+
+# Timecode plot
+maxTimes = [12.29, 25.55, 33.05, 49.15, 63.25, 77.05, 91.05, 105.05, 119.0, 123.05, 137.05]
+
+def getPrefNoBreakpoints(length):  # assuming continuous programme
+    """
+    Return the preferred number of breakpoints from the length in minutes
+    """
+    length = float(length)
+    if length >= 137.05:
+        difference = length - 137.05
+        extra = difference // 14
+        return int(10 + extra)
+
+    else:
+        for i in range(0, len(maxTimes)):
+            if maxTimes[i] > length:
+                return i
+
+
+def roundTime(x, base=5):
+    '''
+    Return x rounded to 5
+    '''
+    return base * round(x / base)
+
+
+def getMidpoint(start, end):
+    '''
+    Return the integer exactly halfway between start and end
+    '''
+    return (start + end) / 2
+
+
+def timecodeToInt(timecode):
+    '''
+    Return timecode in seconds
+    Used in getRuntime function
+    '''
+    temp = timecode.split(':')
+    answer = [int(x) for x in temp]
+
+    hours = (answer[0]-10) * (3600 * 25)
+    mins =answer[1] * 60 * 25
+    secs  =(answer[2] * 25)
+    return hours + mins + secs
+
+def getRuntime(prodNo):
+    '''
+    Return runtime in minutes of a production number
+    Uses ['durations']['runtime'] - different to getLength
+    '''
+    api = f"https://programmeversionapi.prd.bs.itv.com/programmeVersion/{prodNo}"
```

Comment:
We make the same API call quite a few times now so might be worth doing one at the top and using same one each time to save time

Reply:
I passed the json to getRuntime instead

Reply:
can't see that change but ignore me if you haven't pushed it yet